### PR TITLE
Issue #1440 - update get_block in database_api and condenser_api

### PR DESCRIPTION
**Pull request by mariusztrela**
Issue #1440

### CODE REVIEW

File: libraries/plugins/apis/condenser_api/include/steemit/plugins/condenser_api/condenser_api_plugin.hpp [hunk 1]
```diff
@@ -11,7 +12,7 @@ using namespace appbase;
 class condenser_api_plugin : public appbase::plugin< condenser_api_plugin >
 {
 public:
-   APPBASE_PLUGIN_REQUIRES( (steemit::plugins::json_rpc::json_rpc_plugin)(steemit::plugins::database_api::database_api_plugin) )
+   APPBASE_PLUGIN_REQUIRES( (steemit::plugins::json_rpc::json_rpc_plugin)(steemit::plugins::database_api::database_api_plugin)(steemit::plugins::block_api::block_api_plugin) )
```
Comment (mvandeberg) [inline libraries/plugins/apis/condenser_api/include/steemit/plugins/condenser_api/condenser_api_plugin.hpp]: Remove the dependency on block_api_plugin. The point of `api_startup` is to dynamically detect which APIs are loaded without needing a strict dependency. One of the design goals of moving `get_block` to its own API is to disable it if it becomes too expensive of a call.

**Comment by mariusztrela** [comment]
Removing dependency - done.

File: libraries/plugins/apis/condenser_api/condenser_api.cpp [hunk 2]
```diff
@@ -985,13 +991,13 @@ DEFINE_API( condenser_api, get_active_witnesses )
 DEFINE_API( condenser_api, get_block_header )
 {
    CHECK_ARG_SIZE( 1 )
-   return my->_database_api->get_block_header( { args[0].as< uint32_t >() } ).header;
+   return my->_block_api->get_block_header( { args[0].as< uint32_t >() } ).header;
```
Comment (mvandeberg) [inline libraries/plugins/apis/condenser_api/condenser_api.cpp]: assert the block_api exists

File: libraries/plugins/apis/condenser_api/condenser_api.cpp [hunk 3]
```diff
@@ -985,13 +991,13 @@ DEFINE_API( condenser_api, get_active_witnesses )
...
 }
 
 DEFINE_API( condenser_api, get_block )
 {
    CHECK_ARG_SIZE( 1 )
-   return my->_database_api->get_block( { args[0].as< uint32_t >() } ).block;
+   return my->_block_api->get_block( { args[0].as< uint32_t >() } ).block;
```
Comment (mvandeberg) [inline libraries/plugins/apis/condenser_api/condenser_api.cpp]: assert the block_api exists

**Comment by mariusztrela** [comment]
Done

**Pull request merged**